### ISSUE TITLE
Enhancement: Add "parser" field to ParserOptions

### ISSUE THREAD

**Issue by ilyub** [typescript-eslint/typescript-eslint#5328]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Relevant Package

utils

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

I am testing rule that should work with vue components and need to use this config:
```ts
{
  parser: require.resolve("vue-eslint-parser"),
  parserOptions: {
    ecmaFeatures: { jsx: true },
    ecmaVersion: 2017,
    extraFileExtensions: [".vue"],
    parser: "@typescript-eslint/parser",
    project: "./tsconfig.json",
    sourceType: "module"
  }
}
```

This configuration works, but I have to add "// @ts-expect-error" comment

### Fail

```typescript
{
  parser: require.resolve("vue-eslint-parser"),
  parserOptions: {
    ecmaFeatures: { jsx: true },
    ecmaVersion: 2017,
    extraFileExtensions: [".vue"],
    parser: "@typescript-eslint/parser",
    project: "./tsconfig.json",
    sourceType: "module"
  }
}
```

### Pass

```typescript
{
  parser: require.resolve("vue-eslint-parser"),
  parserOptions: {
    ecmaFeatures: { jsx: true },
    ecmaVersion: 2017,
    extraFileExtensions: [".vue"],
    project: "./tsconfig.json",
    sourceType: "module"
  }
}
```

**Comment by bradzacher** [comment]
There are a lot of parser options that we don't support in the types because they don't do anything in our parser.

I think a better course of action would just be to add an unbound string indexer onto the type so that you can pass any additional properties you want.
